plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -287,8 +290,11 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql,
         PreparedStatement statement = connection.prepareStatement(sql);
         // This is a heuristic, not exact science. A better formula can perhaps be found with measurements.
         // Column count is not known for non-SELECT queries. Not setting fetch size for these.
-        if (columnCount.isPresent()) {
-            statement.setFetchSize(max(100_000 / columnCount.get(), 1_000));
+        int fetchSize = jdbcFetchSize.orElseGet(() ->
+                columnCount.map(count -> max(100_000 / count, 1_000))
+                        .orElse(0));
```

Comment:
Can we have it as `Optional<Integer>` and replace he `fetchSize != 0` to `Optional#isPresent` ?